torchvision/models/video/video_trunk.py

```diff
@@ -0,0 +1,191 @@
+import torch
+import torch.nn as nn
+
+from .video_stems import get_default_stem
+
+BLOCK_CONFIG = {
+    10: (1, 1, 1, 1),
+    16: (2, 2, 2, 1),
+    18: (2, 2, 2, 2),
+    26: (2, 3, 4, 3),
+    34: (3, 4, 6, 3),
+    50: (3, 4, 6, 3),
+    101: (3, 4, 23, 3),
+    152: (3, 8, 36, 3)
+}
+
+
+class BasicBlock(nn.Module):
+
+    expansion = 1
+
+    def __init__(self, inplanes, planes, conv_builder, stride=1, downsample=None):
+        midplanes = (inplanes*planes*3*3*3) // (inplanes*3*3 + 3*planes)
+
+        super(BasicBlock, self).__init__()
+        self.conv1 = nn.Sequential(
+            conv_builder(inplanes, planes, midplanes, stride),
+            nn.BatchNorm3d(planes),
+            nn.ReLU(inplace=True)
+        )
+        self.conv2 = nn.Sequential(
+            conv_builder(planes, planes, midplanes),
+            nn.BatchNorm3d(planes)
+        )
+        self.relu = nn.ReLU(inplace=True)
+        self.downsample = downsample
+        self.stride = stride
+
+    def forward(self, x):
+        residual = x
+
+        out = self.conv1(x)
+        out = self.conv2(out)
+        if self.downsample is not None:
+            residual = self.downsample(x)
+ 
+        out += residual
+        out = self.relu(out)
+
+        return out
+
+
+class Bottleneck(nn.Module):
+    expansion = 4
+
+    def __init__(self, inplanes, planes, conv_builder, stride=1, downsample=None):
+
+        super(Bottleneck, self).__init__()
+        midplanes = (inplanes*planes*3*3*3) // (inplanes*3*3 + 3*planes)
+
+        # 1x1x1
+        self.conv1 = nn.Sequential(
+            nn.Conv3d(inplanes, planes, kernel_size=1, bias=False),
+            nn.BatchNorm3d(planes),
+            nn.ReLU(inplace=True)
+        )
+        # Second kernel
+        self.conv2 = nn.Sequential(
+            conv_builder(planes, planes, midplanes, stride),
+            nn.BatchNorm3d(planes),
+            nn.ReLU(inplace=True)
+        )
+
+        # 1x1x1
+        self.conv3 = nn.Sequential(
+            nn.Conv3d(planes, planes * 4, kernel_size=1, bias=False),
+            nn.BatchNorm3d(planes * 4)
+        )
+        self.relu = nn.ReLU(inplace=True)
+        self.downsample = downsample
+        self.stride = stride
+
+    def forward(self, x):
+        residual = x
+
+        out = self.conv1(x)
+        out = self.conv2(out)
+        out = self.conv3(out)
+
+        if self.downsample is not None:
+            residual = self.downsample(x)
+
+        out += residual
+        out = self.relu(out)
+
+        return out
+
+
+class VideoTrunkBuilder(nn.Module):
+
+    def __init__(self, block, conv_makers, model_depth, 
```

Comment:
but that would give us quite a 8-10 constructors for each architecture ... isn't this to an extent cleaner?

Reply:
Well, it depends. When creating a simple training script, we will customize the training script to have a `depth` argument to be passed to the model, but not all models support that, so this will make experimenting with potentially other backbones harder.

But don't worry about this, I can make a pass afterwards and modify a few things.